### PR TITLE
[JENKINS-41348] update UI to use user-scoped credentials

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- See @vivek's instructions from #768 
- See [JENKINS-41348](https://issues.jenkins-ci.org/browse/JENKINS-41348).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
@vivek perhaps you could take a look to validate these changes are appropriate. I tested them and credentials were created correctly in the user's scope and in the correct domain.

File: blueocean-dashboard/src/main/js/creation/credentials/CredentialsApi.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
 import { capabilityAugmenter, Fetch, UrlConfig, Utils } from '@jenkins-cd/blueocean-core-js';
 
+const DOMAIN = 'blueocean-git-domain';
+const SCOPE = 'USER';
+
+function getCredentialsUrl() {
+    const path = UrlConfig.getJenkinsRootURL();
+    return Utils.cleanSlashes(`${path}/blue/rest/organizations/jenkins/credentials/user/`);
```
Comment (vivek) [inline blueocean-dashboard/src/main/js/creation/credentials/CredentialsApi.js]: @cliffmeyers not directly related to this PR but shouldn't base URI should not be hardcoded,  'blue' in /blue/rest/... could change. All of UI should be using appUrlBase set in blueocean-web/main.jsx, https://github.com/jenkinsci/blueocean-plugin/blob/master/blueocean-web/src/main/js/main.jsx#L143.

**Comment by vivek** [comment]
@cliffmeyers there is  hardcoding base url, better to get it from config set in main.jsx, see my comments on this. LGTM otherwise.

File: blueocean-dashboard/src/main/js/creation/credentials/CredentialsApi.js (hunk 1 above)
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/creation/credentials/CredentialsApi.js]: Good catch. We actually handle this somewhat inconsistently in the app and it should be fixed. I will incorporate these changes.

**Comment by vivek** [comment]
@cliffmeyers LGTM 🐝 pending ci and ath runs.

**Comment by michaelneale** [comment]
passes the ATH

**Comment by cliffmeyers** [comment]
ATH passed locally.

```
On branch bug/JENKINS-41348-credential-changes-ui
Your branch is up-to-date with 'origin/bug/JENKINS-41348-credential-changes-ui'.

11:50:17 INFO  - OK. 1079  total assertions passed. (10m 57s)
11:50:17 INFO  - [11:50:17] Finished 'default' after 11 min
**** NightwatchRunner - No failures.
```

**Comment by cliffmeyers** [comment]
ATH passed: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/1515/pipeline/18

**Pull request merged**